### ISSUE TITLE
C# 'Remove Unnecessary Usings' refactoring leads to broken code.

### ISSUE THREAD

**Issue by dotnetjunkie** [dotnet/roslyn#4223]
The new Roslyn compiler incorrectly marks usings as unused for a build configuration where the project setting 'XML documentation file' is unchecked, while the XML comments refer to types in the using list. This appears both in .NET 4.5 and PCL projects. Tested with Visual Studio Community 2015.

Example:

``` c#
using System;

/// <summary><see cref="Type"/>.</summary>
class Class1 { }
```

In a project build where the 'XML documentation file' setting is unchecked, the `using System;` statement will be grayed out. When putting the cursor on it and pressing 'CTRL + .' followed by 'ENTER', the 'Remove Unnecessary Usings' refactoring is executed which removes the using.

Now switch to a build configuration that has  'XML documentation file' setting checked, and you'll see the build failing, because `Type` is referenced from the XML comments, but it's not fully qualified.

In other words, the syntax highlighting and refactoring are broken, because they should take into consideration existing XML comments, even if the current build doesn't use them.

**Comment by bkoelman** [comment]
Same as #4034?

**Comment by dotnetjunkie** [comment]
Yes, you are right. Duplicate of #4034.
